File: services/src/main/java/com/liferay/mobile/sdk/v7/membershiprequest/MembershipRequestService.java
package com.liferay.mobile.sdk.v7.membershiprequest;

import com.liferay.mobile.sdk.Call;
import com.liferay.mobile.sdk.annotation.Param;
import com.liferay.mobile.sdk.annotation.Path;
import com.liferay.mobile.sdk.http.Response;
import java.lang.String;
import org.json.JSONObject;

@Path("/membershiprequest")
public interface MembershipRequestService {
  @Path("/add-membership-request")
  Call<JSONObject> addMembershipRequest(@Param(name = "groupId") long groupId, @Param(name = "comments") String comments, @Param(name = "serviceContext", className = "") JSONObject serviceContext);

  @Path("/delete-membership-requests")
  Call<Response> deleteMembershipRequests(@Param(name = "groupId") long groupId, @Param(name = "statusId") long statusId);

  @Path("/get-membership-request")
  Call<JSONObject> getMembershipRequest(@Param(name = "membershipRequestId") long membershipRequestId);

  @Path("/update-status")
  Call<Response> updateStatus(@Param(name = "membershipRequestId") long membershipRequestId, @Param(name = "reviewComments") String reviewComments, @Param(name = "statusId") long statusId, @Param(name = "serviceContext", className = "") JSONObject serviceContext);
}

File: windows/Liferay.SDK/Service/V70/MembershipRequest/MembershipRequestService.cs
//------------------------------------------------------------------------------
// <copyright file="MembershipRequestService.cs">
//    Copyright (c) 2014-present Andrea Di Giorgi
//
//    Permission is hereby granted, free of charge, to any person obtaining a
//    copy of this software and associated documentation files (the "Software"),
//    to deal in the Software without restriction, including without limitation
//    the rights to use, copy, modify, merge, publish, distribute, sublicense,
//    and/or sell copies of the Software, and to permit persons to whom the
//    Software is furnished to do so, subject to the following conditions:
//
//    The above copyright notice and this permission notice shall be included in
//    all copies or substantial portions of the Software.
//
//    THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
//    IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
//    FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
//    THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
//    LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
//    FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
//    DEALINGS IN THE SOFTWARE.
// </copyright>
// <author>Andrea Di Giorgi</author>
// <website>https://github.com/Ithildir/liferay-sdk-builder-windows</website>
//------------------------------------------------------------------------------

using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Threading.Tasks;

namespace Liferay.SDK.Service.V70.MembershipRequest
{
	public class MembershipRequestService : ServiceBase
	{
		public MembershipRequestService(ISession session)
			: base(session)
		{
		}

		public async Task UpdateStatusAsync(long membershipRequestId, string reviewComments, long statusId, JsonObjectWrapper serviceContext)
		{
			var _parameters = new JsonObject();

			_parameters.Add("membershipRequestId", membershipRequestId);
			_parameters.Add("reviewComments", reviewComments);
			_parameters.Add("statusId", statusId);
			this.MangleWrapper(_parameters, "serviceContext", "$languageUtil.getJSONWrapperClassName($parameter.type)", serviceContext);

			var _command = new JsonObject()
			{
				{ "/membershiprequest/update-status", _parameters }
			};

			await this.Session.InvokeAsync(_command);
		}

		public async Task<dynamic> AddMembershipRequestAsync(long groupId, string comments, JsonObjectWrapper serviceContext)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("comments", comments);
			this.MangleWrapper(_parameters, "serviceContext", "$languageUtil.getJSONWrapperClassName($parameter.type)", serviceContext);

			var _command = new JsonObject()
			{
				{ "/membershiprequest/add-membership-request", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task<dynamic> GetMembershipRequestAsync(long membershipRequestId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("membershipRequestId", membershipRequestId);

			var _command = new JsonObject()
			{
				{ "/membershiprequest/get-membership-request", _parameters }
			};

			var _obj = await this.Session.InvokeAsync(_command);

			return (dynamic)_obj;
		}

		public async Task DeleteMembershipRequestsAsync(long groupId, long statusId)
		{
			var _parameters = new JsonObject();

			_parameters.Add("groupId", groupId);
			_parameters.Add("statusId", statusId);

			var _command = new JsonObject()
			{
				{ "/membershiprequest/delete-membership-requests", _parameters }
			};

			await this.Session.InvokeAsync(_command);
		}
	}
}